docs/reference/commands/reset-elastic-password.asciidoc

```diff
@@ -0,0 +1,57 @@
+[roles="xpack"]
+[[reset-elastic-password]]
+== elasticsearch-reset-elastic-password
+
+The `elasticsearch-reset-elastic-password` command resets the password for the
+`elastic` <<built-in-users,built-in superuser>>.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+bin/elasticsearch-reset-elastic-password
+[-a, --auto] [-b, --batch] [-h, --help]
+[-E <KeyValuePair] [-i, --interactive]
+----
+
+[discrete]
+=== Description
+
+This command resets the password of the `elastic` superuser. By default, a
+strong password is generated for you. If you wish to choose the password, you
+can run the tool in interactive mode with `-i`, where you will be prompted to
+enter the desired password. It uses a temporary superuser in the
+<<file-realm, file realm>> that is automatically generated and then removed,
+in order to submit the request to change the `elastic` user password.
+
+This command uses an HTTP connection to connect to the cluster and run the user
+management requests. The command automatically attempts to establish the connection
+over HTTPS by using the `xpack.security.http.ssl` settings in
+the `elasticsearch.yml` file. If you do not use the default config directory
+location, ensure that the *ES_PATH_CONF* environment variable returns the
+correct path before you run the `elasticsearch-create-enrollment-token` command. You can
+override settings in your `elasticsearch.yml` file by using the `-E` command
+option. For more information about debugging connection failures, see
+<<trb-security-setup>>.
+
+[discrete]
+[[reset-elastic-password-parameters]]
+=== Parameters
+
+`-a, --auto`:: Reset the password of `elastic` to an auto-generated strong password. (Default)
+
+`-b, --batch`:: If enabled, runs the reset password process without prompting the user for verification.
+
+`-i, --interactive`:: When specified, the tool prompts the user for the password of the `elastic` user.
```

Comment:
Could we support `--silent` (and maybe `--verbose`) too? It may be useful to automate password resets, where the new strong password is returned and can be easily saved by the calling script.

Reply:
`--verbose` is supported , I just failed to add it here and in the other doc, will amend

`--silent` wouldn't make much sense here for `--auto` since the user can't get the value that the password is set to. 

> It may be useful to automate password resets, where the new strong password is returned and can be easily saved by the calling script.

Can you clarify what you mean ? Have everything else printed on normal verbosity and then just the password value on `SILENT` so that if you run this with `--silent` you only get the password value in the output ? We can do that but I'm not sure how helpful this will be. I think that 
a) Users scripting this tool will want to select their password or can autogenerate this outside elasticsearch, in the script
b) If one is scripting that, it is not that much  more difficult to parse the line `New value: <password>\n`  than parsing the line `<password>\n`

Reply:
> Users scripting this tool will want to select their password or can autogenerate this outside elasticsearch, in the script

User don't need to care about generating a strong password, and they can rely on the tool itself even if they are running in an automated script. It sounds strange to me to provide this functionality if humans run the tool, but asking a different process if they want to script it.

> If one is scripting that, it is not that much more difficult to parse the line New value: <password>\n than parsing the line <password>\n

If we just print the password, there is no parsing needed, just the output, and it's easy to put in a variable or something like that:

```
PWD=`elasticsearch-reset-elastic-password -b -s`
```

This is way easier than any sort of parsing of a multi-line output message in context where the environment (or the user) haven't options for easy parsing.

I also see that we support those flags in every other CLI too, even if I don't know exactly what they do.

Reply:
>  It sounds strange to me to provide this functionality if humans run the tool, but asking a different process if they want to script it.

We're not asking for a different process. I was just describing my thought process around why I think it is not very helpful. There is nothing preventing the user to use our autogenerated passwords even now. 

> This is way easier than any sort of parsing of a multi-line output message in context where the environment (or the user) haven't options for easy parsing.

If we think from a product perspective that this is something we need, I'm totally fine with it. 

> I also see that we support those flags in every other CLI too, even if I don't know exactly what they do.

I get the argument about consistency but I doubt that these flags are useful in all other tools. For example, one can run
```
bin/elasticsearch-setup-passwords auto --silent
```

which will set all internal users passwords to auto-generated passwords that will not be printed to the console, so users will not know what these are. We probably need to fix that behavior instead of replicating it here.

Reply:
> If we think from a product perspective that this is something we need, I'm totally fine with it.

From a product perspective, It is a nice to have. Not a blocker if it add extra complexity and we want to cut scope or iterate later on that.

> We probably need to fix that behavior instead of replicating it here.

+1 in fixing in other tools. For this specific case, I feel that it makes sense to have since it will print the password. We don't need a `--verbose` flag if there is nothing we want to add to the "standard" output.

Reply:
`--verbose` is supported since we might need to show additional information from the base class. I changed the immplementation to only return the password when run with `--silent`